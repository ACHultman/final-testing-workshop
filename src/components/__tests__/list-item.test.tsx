import { render } from "@testing-library/react";
import ListItem from "../list-item";

describe("listItem component", () => {
  it("matches snapshot", () => {
    const { baseElement } = render(<ListItem text={"test text"} />);
    expect(baseElement).toMatchSnapshot();
  });

  it("inserts placeholder", () => {
    const placeholder = "No entry";
    const { getByText } = render(<ListItem text={undefined} />); // maybe ?? not necessary
    getByText(placeholder);
  });
});
