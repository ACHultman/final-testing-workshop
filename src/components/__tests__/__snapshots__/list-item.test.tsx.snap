// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`listItem component matches snapshot 1`] = `
<body>
  <div>
    <li
      class="list-item"
      data-testid="list-item"
    >
      test text
    </li>
  </div>
</body>
`;
